app/models/student.rb

```diff
@@ -2,4 +2,15 @@ class Student <ApplicationRecord
   has_many :professor_students
   has_many :professors, through: :professor_students
   validates :name, :age, :house, presence: true
+
+
+  def professor_count
+      professors.count
+  end
+
+  def self.average_student_age
+      average(:age)
+  end
```

Comment:
I like to consider where I want to use this information. In order to determine which model a method should live in.  If I am only concerned about displaying the average age of students for a specific professor, I might consider moving this to the professor model. In the professor model this method might looks something like:

```
  def average_student_age
     students.average(:age)
  end
```

Then in the view you would have `@professor.average_student_age` instead of `@professor.students.average_student_age`